apps/dh/api-dh/source/DataHub.WebApi/GraphQL/GraphQLQuery.cs

```diff
@@ -295,10 +295,16 @@ public GraphQLQuery()
                 .Resolve()
                 .WithScope()
                 .WithService<IESettExchangeClient_V1>()
-                .ResolveAsync(async (context, client) =>
+                .WithService<IMarketParticipantClient>()
+                .ResolveAsync(async (context, client, marketParticipantClient) =>
                 {
-                    var documentId = context.GetArgument<string?>("documentId");
-                    return await client.EsettAsync(documentId!);
+                    var gridAreas = await marketParticipantClient.GetGridAreasAsync();
+                    var gridAreaLookup = gridAreas.ToDictionary(x => x.Code);
+                    var documentId = context.GetArgument<string>("documentId");
+                    var exchangeEventTrackignResult = await client.EsettAsync(documentId);
+                    var gridArea = gridAreaLookup[exchangeEventTrackignResult.GridAreaCode];
+                    exchangeEventTrackignResult.GridAreaCode = $"{gridArea.Code} - ${gridArea.Name}";
```

Comment:
suggestion: can see that the same "code - name" format is used in the table view. Right now the table only shows the "code". What if we refactor the code so both views can use it? Better yet,  return "gridAreaCode" and "gridAreaName" as properties for the UI to put the properties together. (the refactor can be in a separate PR).